### PR TITLE
11588 fix permissions values

### PR DESCRIPTION
**Pull request by endulab**
Handle correct values in permissions.

Add a new field AmountInWeis to token criteria dto.
Keep strings instead of floats in token_criteria_item.
HoldingSelectionModel keeps amount in wei-like units but they are displayed in permissions as floats (divided by decimals).
Add decimals to token_list_model.

status-go [pr](https://github.com/status-im/status-go/pull/4727)

Issue #11588

### CODE REVIEW

File: src/app/modules/main/chat_section/module.nim [hunk 1]
```diff
@@ -1320,7 +1319,7 @@ method createOrEditCommunityTokenPermission*(self: Module, communityId: string,
       self.onCommunityTokenPermissionUpdateFailed(communityId)
       return
 
-    tokenCriteriaDto.amount = viewAmount.formatBiggestFloat(ffDecimal)
+    tokenCriteriaDto.amountInWeis = tokenCriteria{"amount"}.getStr
```
Comment (endulab) [inline src/app/modules/main/chat_section/module.nim:1322]: we fill new field

File: ui/app/AppLayouts/Communities/panels/JoinPermissionsOverlayPanel.qml [hunk 2]
```diff
@@ -50,10 +50,6 @@ Control {
         readonly property string channelMembershipRequestPendingText: qsTr("Channel Membership Request Pending...")
         readonly property string memberchipRequestRejectedText: qsTr("Membership Request Rejected")
 
-        function holdingsTextFormat(name, amount) {
```
Comment (endulab) [inline ui/app/AppLayouts/Communities/panels/JoinPermissionsOverlayPanel.qml:53]: unused

File: src/app/modules/main/chat_section/module.nim (hunk 1 above)
Comment (jrainville) [reply src/app/modules/main/chat_section/module.nim:1322]: We can rename it `amountInWei` for it to sound better

File: src/app/modules/main/communities/module.nim [hunk 3]
```diff
@@ -543,6 +543,7 @@ method requestCancelDiscordChannelImport*(self: Module, discordChannelId: string
 
 proc createCommunityTokenItem(self: Module, token: CommunityTokensMetadataDto, communityId: string, supply: string,
     infiniteSupply: bool): TokenListItem =
+  let communityTokenDecimals = if token.tokenType == TokenType.ERC20: 18 else: 0
```
Comment (jrainville) [inline src/app/modules/main/communities/module.nim:546]: Don't tokens come with a `decimals` value sometimes?

**Review by jrainville — COMMENTED** [review]
Nice job. Just two small comments

File: src/app/modules/main/chat_section/module.nim (hunk 1 above)
Comment (endulab) [reply src/app/modules/main/chat_section/module.nim:1322]: Like everywhere? 😱

Comment (jrainville) [reply src/app/modules/main/chat_section/module.nim:1322]: `CTRL+SHIFT+F` -> `Replace all` :grin:

**Pull request merged**